bananas_api/tool_reclassify/reclassify.py

```diff
@@ -0,0 +1,60 @@
+import glob
+import yaml
+
+from .newgrf import load_newgrf
+
+from ..helpers.enums import Availability
+from ..index.common_disk import Index
+from ..new_upload.classifiers.newgrf import Classification, classify_newgrf
+
+
+CLASSIFICATION_TO_FUNCTION = {
+    "newgrf": (load_newgrf, classify_newgrf),
+}
+
+
+def reclassify_and_update_metadata(index_folder, storage_folder, category, unique_id, version):
+    data = Index(index_folder).read_content_version(f"{category}/{unique_id}", version, load_as_object=True)
+
+    # Load the global data to find things like the name.
+    with open(f"{index_folder}/{category}/{unique_id}/global.yaml") as f:
+        global_data = yaml.safe_load(f.read())
+
+    name = data.get("name", global_data.get("name", "Unknown"))
+
+    # Prepare a result object, with most things filled in already.
+    result = {
+        "unique_id": unique_id,
+        "version": version,
+        "md5sum_partial": data["md5sum_partial"],
+        "is_available": data["availability"] == Availability.NEW_GAMES,
+        "name": name,
+        "error": False,
+        "message": "",
+        "classification": Classification.UNKNOWN,
+    }
+
+    # Find the file in the local storage.
+    files = glob.glob(f"{storage_folder}/{category}/{unique_id}/{data['md5sum_partial']}*.tar.gz")
+    if len(files) == 0:
+        result["error"] = True
+        result["message"] = "no matching files in local-storage"
+        return result
+    if len(files) > 1:
+        result["error"] = True
+        result["message"] = "multiple matching files in local-storage"
+        return result
+
+    try:
+        obj = CLASSIFICATION_TO_FUNCTION[category][0](files[0])
+    except Exception as e:
+        result["error"] = True
+        result["message"] = f"error while loading file: {e}"
+        return result
+
+    result["classification"] = CLASSIFICATION_TO_FUNCTION[category][1](obj)
+    data["classification"] = result["classification"]
```

Comment:
Assign to `data` first, copy to `result`. Easier in case we ever want to adjust `result`.